airflow/datasets/manager.py

```diff
@@ -68,6 +68,9 @@ def _queue_dagruns(self, dataset: DatasetModel, session: Session) -> None:
         for dag_id in consuming_dag_ids:
             session.merge(DatasetDagRunQueue(dataset_id=dataset.id, target_dag_id=dag_id))
 
+    def resolve_dataset_model(self, dataset: Dataset) -> DatasetModel:
```

Comment:
i'm not sure resolve is  the right word here.

it's like a mutation hook here.... 

resolve suggests  you are applying some logic to determine which way to go.... but e.g. here you already know it's always gonna be  DatasetModel....  so  what's being  resolved?

Reply:
I'm 100% open to naming suggestions! I didn't love resolve either, but it's also not a mutation 🤷‍♂️

Reply:
right... easier to not like  the naming than to come up with good naming :)

but ... is it not a mutation?

oh cus you're returning a DatasetModel class instead of a Dataset class...

well... i guess we could _make it_ a mutation hook....  and just return a [possibly] mutated Dataset object.... not sure .... maybe @blag will save the day with some brilliance

Reply:
Breaking out my thesaurus...